sdk/core/scheduler/main.cc

```diff
@@ -31,9 +31,10 @@ using namespace CHERI;
 /**
  * Exit simulation, reporting the error code given as the argument.
  */
-void simulation_exit(uint32_t code)
+int simulation_exit(uint32_t code)
```

Comment:
Why is this an `int`?  I don't think any code should ever be expected to handle this returning.  The fallback mode should probably be to infinite loop.

Reply:
It's a cross-compartment call, so `void` trips the compiler's warning.  I'll annotate it as `noreturn` and make it loop, tho'.

Reply:
It's better to silence the warning.  None of the callers gracefully handle it failing in a way that cares about return results, they just continue if they are not in a simulation.  It's always a best-effort thing for debugging.